src/EditorFeatures/Test/Workspaces/ProjectCacheHostServiceFactoryTests.cs

```diff
@@ -136,7 +136,13 @@ public void TestImplicitCacheMonitoring()
             }
             while (weak.IsAlive);
 
-            Assert.False(weak.IsAlive);
+            // FailFast (and thereby capture a dump) to investigate what's
+            // rooting the object.
+            if (weak.IsAlive)
+            {
+                CodeAnalysis.Test.Utilities.ExceptionUtilities.FailFast(new Exception("Please investigate why the object wasn't collected!"));
```

Comment:
The problem we historically had with Environment.FailFast is it's not an "exception" from the Win32 sense -- it just ends in TerminateProcess which isn't really exceptional. This was our old helper to be _really_ exceptional.